File: Wpf/Model/User/UserModel.cs
using Caliburn.Micro;
using Model.Events;

namespace Model.User {
    public class UserModel {
        public UserModel(int id, string firstname, string lastname) {
            Id = id;
            Firstname = firstname;
            Lastname = lastname;
        }

        public int Id { get; private set; }
        public string Firstname { get; private set; }
        public string Lastname { get; private set; }

        public void UpdateDetails(IEventAggregator eventAggregator, string firstname, string lastname) {
            Firstname = firstname;
            Lastname = lastname;
            eventAggregator.Publish(new UserDetailsUpdated(Id));
        }
    }
}

File: Durandal/Model/Model/User/UserModel.js
/*global require, define */
define(['ko', 'durandal/app'], function (ko, eventAggregator) {
    'use strict';
    function UserModel(id, firstName, lastName) {
        this.id = parseInt(id, 10);
        this.firstName = firstName;
        this.lastName = lastName;
        this.updateDetails = this.updateDetails.bind(this);
    }

    UserModel.prototype.updateDetails = function (firstName, lastName) {
        this.firstName = firstName;
        this.lastName = lastName;
        eventAggregator.trigger('userDetailsUpdated', this.id);
    };

    return UserModel;
});